app/controllers/posts_controller.rb

```diff
@@ -30,6 +30,16 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    user = User.find(params[:user_id])
+    user.posts_counter -= 1
+    @post.destroy!
+    user.save
+    flash[:success] = 'Post deleted successfully'
+    redirect_to user_posts_path(user.id)
+  end
+
```

Comment:
- [ ] Great job so far. please kindly use CanCanCan for Post authorization in Post Controller
something like this 
```ruby
 class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```